core/src/tile/labels/labels.cpp

```diff
@@ -32,7 +36,13 @@ bool Labels::addLabel(MapTile& _tile, const std::string& _styleName, Label::Tran
             return false;
         }
 
-        l->update(m_view->getViewProjectionMatrix() * _tile.getModelMatrix(), m_screenSize, 0);
+        // NB: viewOrigin.z is only determined by screen width and height.
+        const auto& viewOrigin = m_view->getPosition();
+
+        auto modelMatrix = glm::scale(glm::mat4(1.0), glm::vec3(_tile.getScale()));
+        modelMatrix[3][2] = -viewOrigin.z;
+
+        l->update(m_view->getViewProjectionMatrix() * modelMatrix, m_screenSize, 0);
```

Comment:
Well, some reason is having the projected position to build geometry out of it. I guess we could remove it, if update is always called before render on tangram api it should be fine. Eventually, this could result in an extra geometry buffer update, i'll check this. Also, I think that this was needed before we fixed concurrency issues and introduce the merge of pending labels from threads to main thread.
@hjanetzek I guess this could work, but feel free to remove the view and this update if it makes things easier!